dotcom-rendering/src/components/MiniProfile.tsx

```diff
@@ -0,0 +1,186 @@
+import { sanitiseHTML } from '../model/sanitise';
+import { palette } from '../palette';
+import type { ServerSideTests, Switches } from '../types/config';
+import type {
+	MiniProfile as MiniProfileModel,
+	StarRating,
+} from '../types/content';
+import { subheadingStyles } from './Subheading';
+
+const miniProfileStyles = css`
+	padding-top: 8px;
+`;
+
+const headingLineStyles = css`
+	width: 140px;
+	margin: 0 0 2px 0;
+	border: none;
+	border-top: 4px solid ${palette('--heading-line')};
+`;
+
+// Nesting is necessary in the bio styles because we receive a string of html from the
+// field. This can contain the following tags:
+// Blocks: p, ul, li
+// Inline: strong, em, a
+const bioStyles = css`
+	${textSans14};
+	padding-top: 6px;
+	color: ${palette('--mini-profiles-text-subdued')};
+	line-height: 1.3rem;
+	p {
+		margin-bottom: 0.5rem;
+	}
+	a {
+		color: ${palette('--caption-link')};
+		text-underline-offset: 3px;
+	}
+	a:not(:hover) {
+		text-decoration-color: ${neutral[86]};
+	}
+	a:hover {
+		text-decoration: underline;
+	}
+	ul {
+		list-style: none;
+		margin: 0 0 0.75rem;
+		padding: 0;
+		margin-bottom: 0.5rem;
+	}
+	ul li {
+		padding-left: 1.25rem;
+	}
+	ul li p {
+		display: inline-block;
+		margin-bottom: 0;
+	}
+	ul li:before {
+		display: inline-block;
+		content: '';
+		border-radius: 0.375rem;
+		height: 10px;
+		width: 10px;
+		margin: 0 0.5rem 0 -1.25rem;
+		background-color: ${palette('--bullet-fill')};
+	}
+	strong {
+		font-weight: bold;
+	}
+`;
+
+const endNoteStyles = css`
+	${textSans14};
+	line-height: 135%;
+	color: ${palette('--mini-profiles-text-subdued')};
+	margin-bottom: 1rem;
+`;
+
+const bottomBorderStyles = css`
+	border-top: 1px solid ${palette('--article-border')};
+	margin-bottom: 0.5rem;
+`;
+
+const headingMarginStyle = css`
+	margin-bottom: 4px;
+`;
+
+interface MiniProfileProps {
+	format: ArticleFormat;
+	ajaxUrl: string;
+	host?: string;
+	pageId: string;
+	isAdFreeUser: boolean;
+	isSensitive: boolean;
+	abTests: ServerSideTests;
+	switches: Switches;
+	editionId: EditionId;
+	hideCaption?: boolean;
+	starRating?: StarRating;
+	miniProfile: MiniProfileModel;
+	RenderArticleElement: ArticleElementRenderer;
+}
+
+export const MiniProfile = ({
+	miniProfile,
+	format,
+	ajaxUrl,
+	host,
+	pageId,
+	isAdFreeUser,
+	isSensitive,
+	switches,
+	abTests,
+	editionId,
+	hideCaption,
+	starRating,
+	RenderArticleElement,
+}: MiniProfileProps) => {
+	return (
+		<>
+			<li css={miniProfileStyles} data-spacefinder-role="nested">
+				<hr css={headingLineStyles} />
+				<h3
+					id={slugify(miniProfile.title)}
+					css={[subheadingStyles(format), headingMarginStyle]}
+				>
+					{miniProfile.title}
+				</h3>
+				<Bio html={miniProfile.bio} />
+				{miniProfile.body.map((element, index) => (
+					<RenderArticleElement
+						// eslint-disable-next-line react/no-array-index-key -- This is only rendered once so we can safely use index to suppress the warning
+						key={index}
```

Comment:
It’s worth noting that [`key={index}` is the React default](https://react.dev/learn/rendering-lists#rules-of-keys), so it’s not really adding anything to make it explicit.

Asserting that a component is only rendered once is also not a real guarantee – in strict mode while developing all elements are rendered twice, and if the mini profile’s body array were to change, it could trigger a re-render, so if there was a unique key it would be better.

Reply:
I didn't know this!

 In the case of the elements that are children of the MiniProfiles `body` - the order in the array should be determined by their order in CAPI so it should remain consistent on each render - it might be better to use that as an explanation. The elements don't have an `id` of any kind though, so without drilling down into what kind of element they are and doing a complicated case statement deriving a key for each one, I'm not sure how to derive one in this case. This will be the case for the other list elements as well

For the MiniProfiles themselves though, the `title` should normally be unique to each one.